src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -212,10 +217,8 @@ def listRunLumis(self, dataset=None, block=None):
                 )
         """
         # Pointless code in python3
-        if isinstance(block, str):
-            block = unicode(block)
-        if isinstance(dataset, str):
-            dataset = unicode(dataset)
+        block = decodeBytesToUnicode(block)
```

Comment:
This code does not make any sense to me! It's also not consistent in this module, e.g.: `listRuns`.
Dario, I'd suggest to completely remove these lines

Reply:
Alan, my job is to make sure that the code that we have right now running in py2 continue to run in the same way in py2 and will continue to be the same in py3.

You are the one that 5y ago added in [this commit](https://github.com/dmwm/WMCore/commit/b50c47968417f7b5321cd595be365a9040a3a6d2) the lines

```python
        # Pointless code in python3
        if isinstance(block, str):
            block = unicode(block)
        if isinstance(dataset, str):
            dataset = unicode(dataset)
```

to `listRunLumis` and not to `listRuns`. Do not complain with me about your inconsistencies.

Moreover, this code in py2 makes sure that if `block` and `dataset` are strings of type sequence of bytes (because `str` in vanilla py2, without any imports from future, means a string of type bytes), then they are converted to unicode strings. `block = decodeBytesToUnicode(block) ` does exactly the same and I had to use it not to rely on the `unicode` keyword, since it is not available in py3 anymore. If you want this code to behave the same in py3 as it is behaving now in py2, this is the only way. 

If you consider that these lines that you added 5 years ago are no longer necessary, then you remove them. It is not my job to refactor your code and decide if it is necessary to explicitely convert to unicode the input of `listRunLumis`, while it is not necessary in the case of `listRuns`. I have to trust that past-you added them for a reason. It is impossible for me to check every time which type a function argument is. I trust that if you decided to convert it from bytes to unicode in py2 explicitely, then
1. you need to continue doing that while we have py2 running
2. you are likely to need it in py3 also

Reply:
You're completely right, Dario! It was an observation (not a complaint by the way) that goes beyond the scope of this code modernization. I need to policy myself and keep focus, sorry about that.
Having a quick look at that commit, it was simply a shift of lines. So the code was already there and now there is no way to know the reason.
Anyways, if we find problems in this module, perhaps removing unnecessary string handling will ease our life. We'll see.

Reply:
Perfect! So, should we keep these changes in this PR, come back to this module with py3 unit tests and eventually moving this discussion to #10293, or did I misunderstood you?

Reply:
Yes, I think we keep the compatibility changes you provided. If we have issues with this module, then I'd say the initial point to look at things might be the previous (and now modernized) unicode casting. I'll make a note to the GH issue you have opened though just to make sure this information doesn't get lost. Thanks again Dario!